### PR TITLE
chore: set up automatic PR labelling

### PR DESCRIPTION
**Pull request by SiTaggart**
When a PR is opened use GitHub actions to automatically label them based on their area

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  - .eslint/**/*
+  - .github/**/*
+  - .jest/**/*
+  - .vscode/**/*
+  - tools/**/*
+
+"Type: Documentation":
+  - packages/paste-website/src/pages/**/*
+
+"Type: Tests":
```
Comment (gloriliale) [inline .github/labeler.yml:32]: Does this mean that there can be both a Type and Area label for a PR?

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# https://github.com/actions/labeler
+
+# Add 'repo' label to any root file changes
+"Area: Repo":
+  - "*"
+
+"Area: Components":
+  - packages/paste-core/components/**/*
```
Comment (TheSisb) [inline .github/labeler.yml:8]: Do we want coverage over layout and primitive components? or core?

**Review by TheSisb — APPROVED** [review]
Pretty cool!

**Review by gloriliale — APPROVED** [review]
Just had a question about the number of labels you can put on a PR, but this is exciting!

**Review by gloriliale — COMMENTED** [review]
Question: why is this a separate PR from the mergify one? Don't we have this same info in the mergify PR?

File: .github/labeler.yml (hunk 1 above)
Comment (SiTaggart) [reply .github/labeler.yml:32]: Yup, you can have as many labels as you like

File: .github/labeler.yml (hunk 2 above)
Comment (SiTaggart) [reply .github/labeler.yml:8]: Oh, maybe. One for each?

- `Area: Components`
- `Area: Primitives`
- `Area: Layout`
- `Area: Core bundle`

**Comment by SiTaggart** [comment]
> Question: why is this a separate PR from the mergify one? Don't we have this same info in the mergify PR?
 
@gloriliale Yeah I mentioned this is in the review request in Slack. I set this up for the mergify PR, but I'm unsure how much of a good idea allowing a third party merge rights is long term. But the labelling seems like it's worth saving. So i created this PR by cherry picking the commit

**Pull request merged**